src/test/rustdoc-ui/reference-link-reports-error-once.rs

```diff
@@ -0,0 +1,20 @@
+#![deny(broken_intra_doc_links)]
+
+/// Links to [a] [link][a]
+/// And also a [third link][a]
+/// And also a [reference link][b]
+///
+/// Other links to the same target should still emit error: [ref] //~ERROR unresolved link to
+/// Duplicate [ref] //~ERROR unresolved link to
+///
+/// Other links to other targets should still emit error: [ref2] //~ERROR unresolved link to
+/// Duplicate [ref2] //~ERROR unresolved link to
+///
+/// [a]: ref
+//~^ ERROR unresolved link to
+/// [b]: ref2
```

Comment:
In the current patch, if this is `[b]: ref`, the error emitted for `[reference link][b]` will point to the definition of `a`. This is because the link cache ignores the name of the reference and handles a and b as if identical. This isn't ideal, but I consider this acceptable: we don't report all errors, but if the user fixes link `a`, we will report an error for link `b`, so eventually, every error can be caught :)

We could fix this by adding the span to `ResolutionInfo` (Optionally, only for Shortcut/Reference links), but I'm not sure if the complexity is worth it.

Reply:
Wait, I'm confused what those have to do with each other. We never cache the span, so there's nowhere for it to look up the span of `a` in the first place, the only span *available* is that of `b`.

Can you post the error that's wrong?

Reply:
Assume the following:

```
[a] [b]

[a]: ref
[b]: ref
```

We would only report an error for `a`.